cloudml-template/examples/babyweight-custom/trainer/task.py

```diff
@@ -146,6 +144,18 @@ def initialise_hyper_params(args_parser):
         default=0.1,
         type=float
     )
+    args_parser.add_argument(
+        '--learning-rate-decay-factor',
+        help="""\
+             **VALID FOR CUSTOM MODELS**
+             The factor by which the learning rate should decay by the end of the training.
+             decayed_learning_rate = learning_rate * decay_rate ^ (global_step / decay_steps)
+             If set to 1.0 (default), then no decay will occur
+             If set to 0.5, then the learning rate should reach 0.5 of its original value  \
```

Comment:
OK, given this formula (defined in the [docs](https://www.tensorflow.org/versions/r1.4/api_docs/python/tf/train/exponential_decay)): 

decayed_learning_rate = learning_rate * decay_rate ^ (global_step / decay_steps)

if the decay_rate=1, then learning_rate will still be the same, regardless of the value of the global_step or decay step, since 1 the power anything=1


if the decay_rate=0.5, then the learning rate will **reach** 0.5 of its original value at the end of the training. Please not that, global_step will always be **less than or equals** to decay_steps, since the decay_steps is set to train_steps

- in the update_learning_rate function in model.py, I have set:

    decay_steps = task.HYPER_PARAMS.train_steps  # decay after each training step

instead of:

    decay_steps = task.HYPER_PARAMS.num_epochs  # decay after each epoch

- I have also improved the description of the parameter in the task.py. as follows:

"If set to 0.5, then the learning rate should reach 0.5 of its original value at the end of the training. Note that, the decay_steps is set to train_steps"